### DISCUSSION TITLE
`rabbitmq_delayed_message_exchange` is not compatible with 3.13.0 and `khepri_db` experimental feature

### DISCUSSION THREAD

**Discussion by pnchen** [rabbitmq/rabbitmq-delayed-message-exchange#271] [General]
### Describe the bug

Error Info:
```
# rabbitmq-plugins enable rabbitmq_delayed_message_exchange
Enabling plugins on node rabbit@debian:
rabbitmq_delayed_message_exchange
The following plugins have been configured:
  rabbitmq_delayed_message_exchange
  rabbitmq_management
  rabbitmq_management_agent
  rabbitmq_mqtt
  rabbitmq_web_dispatch
  rabbitmq_web_mqtt
Applying plugin configuration to rabbit@debian...
Stack trace:

** (CaseClauseError) no case clause matching: {:badrpc, {:EXIT, {:aborted, {:node_not_running, :rabbit@debian}}}}
    (rabbitmqctl 3.13.0-dev) lib/rabbitmq/cli/plugins/plugins_helpers.ex:215: RabbitMQ.CLI.Plugins.Helpers.update_enabled_plugins/2
    (rabbitmqctl 3.13.0-dev) lib/rabbitmq/cli/plugins/plugins_helpers.ex:112: RabbitMQ.CLI.Plugins.Helpers.update_enabled_plugins/4
    (rabbitmqctl 3.13.0-dev) lib/rabbitmq/cli/plugins/commands/enable_command.ex:127: anonymous fn/6 in RabbitMQ.CLI.Plugins.Commands.EnableCommand.do_run/2
    (elixir 1.15.4) lib/stream.ex:1530: anonymous fn/2 in Stream.iterate/2
    (elixir 1.15.4) lib/stream.ex:1757: Stream.do_unfold/4
    (elixir 1.15.4) lib/stream.ex:1828: Enumerable.Stream.do_each/4
    (elixir 1.15.4) lib/stream.ex:1052: Stream.do_transform_inner_enum/7
    (elixir 1.15.4) lib/stream.ex:1828: Enumerable.Stream.do_each/4

{:case_clause, {:badrpc, {:EXIT, {:aborted, {:node_not_running, :rabbit@debian}}}}}
```

the node `rabbit@debian` is running, and plugins such as `rabbitmq_mqtt`  has been enabled succesfully
![image](https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/assets/2562679/9c817738-c3be-4148-a991-ac45e1912361)

### Reproduction steps

- install the latest version of rabbitmq, follow https://www.rabbitmq.com/docs/install-debian#apt-quick-start-cloudsmith
- run `rabbitmq-plugins enable rabbitmq_delayed_message_exchange`

### Expected behavior

enable the plugin without error

**Comment by mkuratczyk** [comment]
You are trying to enable a plugin while RabbitMQ is not running. Start RabbitMQ first.

**Comment by pnchen** [comment]
> You are trying to enable a plugin while RabbitMQ is not running. Start RabbitMQ first.

Thanks! In my Environment, The RabbitMQ is already started And `rabbitmq_mqtt` can be enabled successfully.

if I stop RabbitMQ by running `systemctl stop rabbitmq-server`, then run `rabbitmq-plugins enable rabbitmq_delayed_message_exchange` it will show
```
Enabling plugins on node rabbit@debian:
rabbitmq_delayed_message_exchange
The following plugins have been configured:
  rabbitmq_delayed_message_exchange
  rabbitmq_management
  rabbitmq_management_agent
  rabbitmq_mqtt
  rabbitmq_web_dispatch
  rabbitmq_web_mqtt
Applying plugin configuration to rabbit@debian...
The following plugins have been enabled:
  rabbitmq_delayed_message_exchange

set 6 plugins.
Offline change; changes will take effect at broker restart.
```

But I can't start the RabbitMQ again , because rabbitmq_delayed_message_exchange will error at starting

**Comment by lukebakken** [comment]
Something similar was reported here - https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/discussions/269

@pnchen you _MUST_ provide the _EXACT_ steps you used to download and install this plugin. Please also provide the output of `ls -la` on the plugins directory in your environment.

**Comment by pnchen** [comment]
I found it because I enable the Feathure flag `khepri_db`.  When I keep this disable every thing will OK. 
<img width="887" alt="截屏2024-03-13 22 03 28" src="https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/assets/2562679/a97291e6-e532-4916-a8bc-ff1a2496c6ad">

Thanks to All

**Comment by michaelklishin** [comment]
@pnchen please never used issues for questions in repositories that have discussions enabled. This is just rude at this point, discussions have been around for a couple of years.

**Comment by michaelklishin** [comment]
This plugin is not compatible with Khepri because when Khepri is enabled, Mnesia is likely not even started, and this plugin uses Mnesia exclusively right now.

If someone wants it to be compatible, you are welcome to contribute a PR.

**Comment by michaelklishin** [comment]
As I was putting together an issue about Khepri support, I have realized a couple of things. This plugin cannot realistically use Khepri:

1. Khepri has a documented limitation where it stores the entire data set both on disk and in memory
2. This plugin stores the entire message content in the table, therefore, it explicitly goes against the recommendation for the current version of Khepri

Given that #229 will take a long time to finish regardless of the details, perhaps this plugin should be reworked to not store the payload in Mnesia. DETS is one readily available option with the known 4 GiB limit per file :(

**Reply by pnchen** [reply]
Sorry, I don't need to use Khepri. I misunderstand  the Rabbitmq's warning At the top of management page ,and enabled all the  feature flags.
Thanks for your reply
